Pal/src/host/FreeBSD/db_misc.c

```diff
@@ -199,54 +199,33 @@ static int init_randgen (void)
     _DkInternalUnlock(&lock);
     return 0;
 }
-int getrand (void * buffer, int size)
+
+void getrand (void * buffer, size_t size)
 {
     unsigned long val;
-    int bytes = 0;
+    size_t bytes = 0;
 
-    _DkInternalLock(&lock);
-    while (!randval) {
-        _DkInternalUnlock(&lock);
-        if (init_randgen() < 0)
-            return -PAL_ERROR_DENIED;
-        _DkInternalLock(&lock);
-    }
+    int ret = init_randgen();
```

Comment:
This should be probably locked. Also, if we are doing it for every `getrand` call, then why `randval` is global?